DOCUMENTATION.md

```diff
@@ -3124,6 +3125,55 @@ let customHandler (dataObj : obj) : HttpHandler =
         // ... do more...
 ```
 
+## Testing
+
+Testing a Giraffe application builds on [ASP.NET Core testing](https://docs.microsoft.com/en-us/aspnet/core/test/middleware?view=aspnetcore-3.1).
+
+### Necessary imports:
+```fsharp
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.TestHost
+open Microsoft.AspNetCore.Hosting
+```
+
+### Build a test host:
+```fsharp
+let getTestHost =
+    WebHostBuilder()
+        .UseTestServer()
+        .Configure(Action<IApplicationBuilder> [YourApp].configureApp)
+        .ConfigureServices([YourApp].configureServices)
+        .ConfigureLogging([YourApp].configureLogging)
+        .UseUrls([YourUrl])
+```
+
+### Build a test server:
+```fsharp
+let getTestServer =
+    let testHost = getTestHost
+    new TestServer(testHost)
+```
+
+### Build a test client:
+```fsharp
+let getTestClient =
```

Comment:
Changing as per suggestions :-).

Reply:
So it turns out (https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/) that "use" here isn't appropriate. Reverting. Runs fine - disposing leads to `Cannot access a disposed object`.

Reply:
Thanks for responding to the feedback, it's a bit interesting in that although this is a HttpClient, it's not a "real" one, so there are no sockets involved for example.

The reason for the object disposed exception will likely be because the client is being reused, I suggested adding `()` to the binding and calls, so it's actually a function that gets a new one each time (if that makes sense).

Disposing of the `HttpClient` I suspect won't dispose of the `TestServer` which it was created from, and that's the bit I'm not sure about with this whole PR. If you look at our own tests, we solve it like this:
https://github.com/giraffe-fsharp/Giraffe/blob/master/tests/Giraffe.Tests/Helpers.fs#L185
and tests look like this:
https://github.com/giraffe-fsharp/Giraffe/blob/master/tests/Giraffe.Tests/PreconditionalTests.fs#L193-L209

i.e. a function a that owns the `TestServer` and disposes it, you pass in a request and it gives you a response. Not saying that this is better, but satisfies any concerns about not disposing of `TestServer`, which I suspect it something that should be done.

Reply:
Thanks for the detailed feedback! I'll dig into this and make some adjustments, then re-push.